pkg/controller/datavolume-controller.go

```diff
@@ -258,8 +258,12 @@ func (r *DatavolumeReconciler) Reconcile(req reconcile.Request) (reconcile.Resul
 	}
 
 	if !pvcExists {
+		cloneStrategy, err := r.getCloneStrategy()
+		if err != nil {
+			return reconcile.Result{}, err
+		}
 		snapshotClassName, err := r.getSnapshotClassForSmartClone(datavolume)
-		if err == nil {
+		if (err == nil) && (cloneStrategy == cdiv1.CloneStrategySnapshot) {
```

Comment:
Any reason for the ( .. ) && ( .. ) the equals have higher priority so there is no risk it won't work without the parenthesis.